### PR TITLE
Native memory management documentation

### PR DESCRIPTION
**Pull request by zakkak**
Documentation follow up work for https://github.com/quarkusio/quarkus/issues/28267.

Add native memory management section to the native reference guide. As per [discussion on quarkus-dev](https://groups.google.com/g/quarkus-dev/c/6nQ7TdXh3I8/m/2DXyJ8mZAQAJ), it was suggested to put this content in the reference guide, rather than create a different document.

There's some indentation changes for existing content to accommodate the new content.

I will squash commits when the review has completed.

Supersedes: #28628

### CODE REVIEW

File: docs/src/main/asciidoc/native-reference.adoc [hunk 1]
```diff
@@ -12,14 +12,157 @@ xref:building-native-image.adoc[Building a Native Executable],
...
+
+* <<native-memory-management,Native Memory Management>>
+* <<inspecting-and-debugging,Inspecting and Debugging Native Executables>>
+* <<native-faq,Frequently Asked Questions>>
+
+[[native-memory-management]]
+== Native Memory Management
+Memory management for Quarkus native executables is enabled by GraalVM’s SubstrateVM runtime system.
+The memory management component in GraalVM is explained in detail
+link:https://www.graalvm.org/22.2/reference-manual/native-image/optimizations-and-performance/MemoryManagement[here].
```
Comment (galderz) [inline docs/src/main/asciidoc/native-reference.adoc:27]: `22.2` should be `{graalvm-version}`?

File: docs/src/main/asciidoc/native-reference.adoc [hunk 2]
```diff
@@ -12,14 +12,157 @@ xref:building-native-image.adoc[Building a Native Executable],
...
+To build Quarkus native with epsilon GC, pass the following argument at build time:
+
+[source,bash]
+----
+-Dquarkus.native.additional-build-args=--gc=epsilon
+----
+
+=== Memory Management Options
+Options to control maximum heap size, young space and other typical use cases found in the JVM can be found in
+https://www.graalvm.org/22.2/reference-manual/native-image/optimizations-and-performance/MemoryManagement[the GraalVM memory management guide].
```
Comment (galderz) [inline docs/src/main/asciidoc/native-reference.adoc:98]: Here as well.

**Review by galderz — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/native-reference.adoc (hunk 1 above)
Comment (zakkak) [reply docs/src/main/asciidoc/native-reference.adoc]: Correct, thanks for catching those.

**Review by gsmet — APPROVED** [review]

**Review by galderz — APPROVED** [review]

**Comment by zakkak** [comment]
@gsmet if this is going to be backported I guess we will need to backport https://github.com/quarkusio/quarkus/pull/28843 as well.

**Pull request merged**